### ISSUE TITLE
user-agent passed by ytdl not used by mpv

### ISSUE THREAD

**Issue by selfisekai** [mpv-player/mpv#8786]
### Important Information

Provide following Information:
- mpv version: 0.33.1
- Platform and Version: Void Linux (glibc)
- Source of the mpv binary: Void Linux's xbps repo

If you're not using git master or the latest release, update.
Releases are listed here: https://github.com/mpv-player/mpv/releases

### Reproduction steps

Try to point mpv to a fragmented video via ytdl, like this: `mpv ytdl://https://selfisekai.rocks/test.m3u8`

### Expected behavior

mpv should use the User-Agent as specified by ytdl (`.requested_formats[].http_headers`):
```json
{
  "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3717.1 Safari/537.36",
  "Accept-Charset": "ISO-8859-1,utf-8;q=0.7,*;q=0.7",
  "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8",
  "Accept-Encoding": "gzip, deflate",
  "Accept-Language": "en-us,en;q=0.5"
}
```

### Actual behavior

nginx logs:
```
92.168.250.204 - - [28/Apr/2021:01:33:43 +0200] "HEAD /test.m3u8 HTTP/1.1" 200 0 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3717.1 Safari/537.36"
192.168.250.204 - - [28/Apr/2021:01:33:43 +0200] "GET /test.m3u8 HTTP/1.1" 200 1148 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3717.1 Safari/537.36"
192.168.250.204 - - [28/Apr/2021:01:33:43 +0200] "GET /f08e80da-bf1d-4e3d-8899-f0f6155f6efa_video_1080_4800000.m3u8 HTTP/1.1" 404 146 "-" "libmpv"
```
This actually breaks playing from some websites, at least https://pilot.wp.pl.

### Log file

Seems useless but here you go: https://0x0.st/-mnE.txt

### Sample files

n/a

**Comment by avih** [comment]
> Expected behavior
> mpv should use the User-Agent as specified by ytdl

No, it should not. mpv only uses ytdl to get the stream URL. Once it has the URL, it's ffmpeg which performs the network access and reads the stream. ytdl is not used to read the stream itself.

**Comment by TheAMM** [comment]
> ytdl is not used to read the stream itself.

That's the point. youtube-dl gives mpv a User-Agent to use, and mpv does not use it. As stated in the issue report, this breaks youtube-dl functionality on sites that expect a specific User-Agent. ytdl_hook should handle this - hopefully youtube-dl only returns a User-Agent when it's necessary, so the user's configured UA is used in other cases.

**Comment by ChrisK2** [comment]
Setting the user-agent [is actually implemented](https://github.com/mpv-player/mpv/blob/master/player/lua/ytdl_hook.lua#L106-L109), but I guess youtube-dl's extractor doesn't think it needs to be set, I guess. Or some other bug breaks it.

**Comment by selfisekai** [comment]
Seems like the code uses headers from `.http_headers` if it's a single format (https://github.com/mpv-player/mpv/blob/master/player/lua/ytdl_hook.lua#L558), but in case of 2 separate formats (video+audio) nothing is done. Each format can have other headers to set - they're set in `.requested_formats[].http_headers`.
